src/test/java/uk/nhs/digital/nhsconnect/lab/results/mesh/message/WorkflowIdTest.java

```diff
@@ -3,20 +3,34 @@
 import org.junit.jupiter.api.Test;
 import uk.nhs.digital.nhsconnect.lab.results.mesh.exception.MeshWorkflowUnknownException;
 
+import static org.junit.jupiter.api.Assertions.assertAll;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class WorkflowIdTest {
 
     @Test
     void testFromStringReturnsWorkflowIdForValidWorkflowIdString() {
-        assertEquals(WorkflowId.REGISTRATION, WorkflowId.fromString(WorkflowId.REGISTRATION.getWorkflowId()));
+        assertAll(() -> {
+            assertEquals(WorkflowId.PATHOLOGY, WorkflowId.fromString(WorkflowId.PATHOLOGY.getWorkflowId()));
+            assertEquals(WorkflowId.PATHOLOGY_ACK, WorkflowId.fromString(WorkflowId.PATHOLOGY_ACK.getWorkflowId()));
+            assertEquals(WorkflowId.SCREENING, WorkflowId.fromString(WorkflowId.SCREENING.getWorkflowId()));
+            assertEquals(WorkflowId.SCREENING_ACK, WorkflowId.fromString(WorkflowId.SCREENING_ACK.getWorkflowId()));
+        });
     }
 
     @Test
     void testFromStringReturnsWorkflowIdForValidLowercaseWorkflowIdString() {
-        final String lowercaseWorkflowId = WorkflowId.REGISTRATION.getWorkflowId().toLowerCase();
-        assertEquals(WorkflowId.REGISTRATION, WorkflowId.fromString(lowercaseWorkflowId));
+        assertAll(() -> {
+            assertEquals(WorkflowId.PATHOLOGY,
+                WorkflowId.fromString(WorkflowId.PATHOLOGY.getWorkflowId().toLowerCase()));
+            assertEquals(WorkflowId.PATHOLOGY_ACK,
+                WorkflowId.fromString(WorkflowId.PATHOLOGY_ACK.getWorkflowId().toLowerCase()));
+            assertEquals(WorkflowId.SCREENING,
+                WorkflowId.fromString(WorkflowId.SCREENING.getWorkflowId().toLowerCase()));
+            assertEquals(WorkflowId.SCREENING_ACK,
+                WorkflowId.fromString(WorkflowId.SCREENING_ACK.getWorkflowId().toLowerCase()));
+        });
```

Comment:
This will still stop at the first failure
```suggestion
        assertAll(
            () -> assertEquals(WorkflowId.PATHOLOGY,
                WorkflowId.fromString(WorkflowId.PATHOLOGY.getWorkflowId().toLowerCase())),
            () -> assertEquals(WorkflowId.PATHOLOGY_ACK,
                WorkflowId.fromString(WorkflowId.PATHOLOGY_ACK.getWorkflowId().toLowerCase())),
            () -> assertEquals(WorkflowId.SCREENING,
                WorkflowId.fromString(WorkflowId.SCREENING.getWorkflowId().toLowerCase())),
            () -> assertEquals(WorkflowId.SCREENING_ACK,
                WorkflowId.fromString(WorkflowId.SCREENING_ACK.getWorkflowId().toLowerCase()))
        });
```